components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/DataFlowProcessingEngine.cs

```diff
@@ -126,18 +119,37 @@ public class DataFlowProcessingEngine : IProcessingEngine, IDisposable {
                     });
 
                 _sinkBlock = new ActionBlock<NetworkMessageModel[]>(
-                    async input => await _messageSink.SendAsync(input).ConfigureAwait(false),
+                    async input => {
+                        await _messageSink.SendAsync(input).ConfigureAwait(false);
+                        if (_batchTriggerInterval > TimeSpan.Zero) {
+                            _batchTriggerIntervalTimer.Change(_batchTriggerInterval, _batchTriggerInterval);
+                        }
+                    },
                     new ExecutionDataflowBlockOptions {
-                        CancellationToken = cancellationToken
+                        CancellationToken = cancellationToken,
                     });
                 _batchDataSetMessageBlock.LinkTo(_encodingBlock);
                 _encodingBlock.LinkTo(_batchNetworkMessageBlock);
                 _batchNetworkMessageBlock.LinkTo(_sinkBlock);
 
+                _messageTrigger.OnMessage += MessageTriggerMessageReceived;
+                if (_diagnosticInterval > TimeSpan.Zero) {
+                    _diagnosticsOutputTimer.Change(_diagnosticInterval, _diagnosticInterval);
+                }
+
                 await _messageTrigger.RunAsync(cancellationToken).ConfigureAwait(false);
+
             }
             finally {
                 IsRunning = false;
+                _messageTrigger.OnMessage -= MessageTriggerMessageReceived;
+                _diagnosticsOutputTimer.Change(Timeout.Infinite, Timeout.Infinite);
+                _batchTriggerIntervalTimer.Change(Timeout.Infinite, Timeout.Infinite);
+                await Task.WhenAll(
```

Comment:
Isnt it enough to wait on the sink block?

Reply:
most probably probably it is